src/generator.cs

```diff
@@ -4371,9 +4370,8 @@ void GenerateProperty (Type type, PropertyInfo pi, List<string> instance_fields_
 				print ("}");
 			}
 			if (pi.CanWrite) {
-				PrintPlatformAttributes (pi);
-				PrintPlatformAttributes (pi.GetSetMethod ());
-				PrintPreserveAttribute (pi.GetSetMethod ());
+				PrintAttributes (pi, platform:true);
```

Comment:
That's not fully correct (it can print different or duplicate data) but it's not related to this change. I filled https://bugzilla.xamarin.com/show_bug.cgi?id=53508